common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java

```diff
@@ -72,6 +72,21 @@ public class ContainerSearchFeature extends Feature {
     @Persisted
     public final Config<Boolean> filterInPetMenu = new Config<>(true);
 
+    @Persisted
```

Comment:
Is there no way we can make this more general? Most of the other code is fine to improve in steps, but adding a bunch of settings mean that any kind of future improvement to this will require an upfixer.

But then again our settings type system might be too limited to make this better. :(

Reply:
One thing we could do would be to instead have a config like: filterInMenues = { ALL, NONE, SOME }, and if it is set to SOME, we can have a String menusToFilterIn = "content_book, bookshelf" or something like that, which we parse as being a space and/or comma-separated list of the enums.

I imagine most people will want to have it at ALL anyway. If they do, when a new menu type is added, it will be supported automatically. If you prefer to only have it in a few, or none, when a new menu type is added, it will be off by default if you have NONE or SOME.

A somewhat more complex solution would be to have ALL, NONE, INCLUDE, EXCLUDE and a menusFilterList, where INCLUDE would work as SOME above, but EXCLUDE would mean "all but those in the menusFilterList".

Even more ideally, we would add some GUI support for creating the menusFilterList. That, I think, would be the ideal way of doing this.

Reply:
Hm. Basically, I think I'm saying that we should support Set<Enum> as a valid configuration type in the GUI. I don't think that would be too hard to do, and it will probably help us elsewhere.

Reply:
We could also have a "list selection" config that could have a supplier for the allowed values.

Reply:
When I was redesigning the settings screen I was thinking about different config types and lists was one of them. It could have an button labelled "Edit" which would then replace all the listed config options with all the inputs for the list, say bank quick jumps for example, it would have "Account Bank Quick Jumps" with an edit button, clicking that would then show 6 input boxes to enter the 6 destinations, or the ranged slider if we were to add that back